contrib/kube-prometheus/docs/How-to-monitor-an-external-secure-etcd-cluster.md

```diff
@@ -0,0 +1,164 @@
+# How to monitor a secure external etcd service with Prometheus Operator
+This guide will help you monitor an external etcd cluster. When the etcd is not hosted inside Kubernetes.
+This is often the case with the Kubernetes setup. This has been tested with kube-aws but same principals will apply to other tools.
+
+# Step 1 - Make the etcd certificates available to Prometheus pod
+Prometheus Operator (and Prometheus) allow us to specify a tlsConfig. This is required as most likely your etcd metrics end points is secure.
+
+## a - Create the secrets in the namespace 
+Prometheus Operator allows us to mount secrets in the pod. By loading the secrets as files, they can be made available inside the Prometheus pod.
+
+`kubectl -n monitoring create secret generic etcd-certs --from-file=CREDENTIAL_PATH/etcd-client.pem --from-file=CREDENTIAL_PATH/etcd-client-key.pem --from-file=CREDENTIAL_PATH/ca.pem`
+
+where CREDENTIAL_PATH is the path to your etcd client credentials on your work machine. 
+(Kube-aws stores them inside the credential folder).
+
+## b - Get Promnetheus Operator to load the secret
+In the previous step we have named the secret 'etcd-certs'.
+
+Edit prometheus-operator/contrib/kube-prometheus/manifests/prometheus/prometheus-k8s.yaml and add the secret under the spec of the Prometheus object manifest:
+
+```
+  secrets: 
+  - etcd-certs
+```
+
+The manifest will look like that:
+```
+apiVersion: monitoring.coreos.com/v1alpha1
+kind: Prometheus
+metadata:
+  name: k8s
+  labels:
+    prometheus: k8s
+spec:
+  replicas: 2
+  secrets: 
+  - etcd-certs
+  version: v1.7.0
+```
+
+If your Prometheus Operator is already in place, update it:
+
+`kubectl -n monitoring replace -f contrib/kube-prometheus/manifests/prometheus/prometheus-k8s.yaml
+
+# Step 2 - Create the Service, endpoints and ServiceMonitor
+
+The below manifest creates a Service to expose etcd metrics (port 2379)
+Replace IP_OF_YOUR_ETCD_NODE_[0/1/2] with the IP addresses of your etcd nodes. If you have more than one node, add them to the same list.
+
+In this example we use insecureSkipVerify: true as kube-aws default certiicates are not valid against the IP. They were created for the DNS. Depending on your use case, you might want to remove this flag or set it to false. (true required for kube-aws if using default certificate generators method)
```

Comment:
Can you check what alternate names there are in the certificates? Because you might be able to actually verify them by setting the `serverName` setting in the `tlsConfig` if there is an alternate name that you can use.

Reply:
no luck here with default kube-aws generated certificate. I wasn't sure how to retrieve the alternate name if any. It would have to be valid for my 3 nodes. For my future cluster, generating my own certificates, this won't be an issue as I can specify an alternate name. I added some comments to indicate either can be used

Reply:
You can check out the alternate names with 

```
openssl x509 -text -in server.crt
```

If I recall correctly, the etcd certs from kube-aws actually did set a generic alternate name, but I could be wrong.